vendor/diff-parsers-upstream-std.patch

```diff
@@ -0,0 +1,301 @@
+-        pp f "@[<2>val @ %a%a%s@ :@ %a@]%a"
+-          mutable_flag mf virtual_flag vf s.txt (core_type ctxt) ct
+-          (item_attributes ctxt) x.pctf_attributes
+-    | Pctf_method (s, pf, vf, ct) ->
+-        pp f "@[<2>method %a %a%s :@;%a@]%a"
+-          private_flag pf virtual_flag vf s.txt (core_type ctxt) ct
+-          (item_attributes ctxt) x.pctf_attributes
+-    | Pctf_constraint (ct1, ct2) ->
+-        pp f "@[<2>constraint@ %a@ =@ %a@]%a"
+-          (core_type ctxt) ct1 (core_type ctxt) ct2
+-          (item_attributes ctxt) x.pctf_attributes
+-    | Pctf_attribute a -> floating_attribute ctxt f a
+-    | Pctf_extension e ->
+-        item_extension ctxt f e;
+-        item_attributes ctxt f x.pctf_attributes
+-  in
+   pp f "@[<hv0>@[<hv2>object@[<1>%a@]@ %a@]@ end@]"
+     (fun f -> function
+          {ptyp_desc=Ptyp_any; ptyp_attributes=[]; _} -> ()
+        | ct -> pp f " (%a)" (core_type ctxt) ct) ct
+-    (list class_type_field ~sep:"@;") l
++    (list (class_type_field ctxt) ~sep:"@;") l
+ 
+ (* call [class_signature] called by [class_signature] *)
+ and class_type ctxt f x =
+   match x.pcty_desc with
+   | Pcty_signature cs ->
+@@@@
+   | Pwith_modsubst (li, li2) ->
+       pp f "module %a :=@ %a" longident_loc li longident_loc li2
+   | Pwith_modtypesubst (li, mty) ->
+       pp f "module type %a :=@ %a" longident_loc li (module_type ctxt) mty;
+ 
+-
+ and module_type1 ctxt f x =
+   if x.pmty_attributes <> [] then module_type ctxt f x
+   else match x.pmty_desc with
+     | Pmty_ident li ->
+         pp f "%a" longident_loc li;
+@@@@
+         pp f "(%a)(%a)" (module_expr ctxt) me1 (module_expr ctxt) me2
+         (* Cf: #7200 *)
+     | Pmod_unpack e ->
+         pp f "(val@ %a)" (expression ctxt) e
+     | Pmod_extension e -> extension ctxt f e
++    | Pmod_hole ->
++        pp f "_"
+ 
+ and structure ctxt f x = list ~sep:"@\n" (structure_item ctxt) f x
+ 
+ and payload ctxt f = function
+   | PStr [{pstr_desc = Pstr_eval (e, attrs)}] ->
+@@@@
+ let core_type = core_type reset_ctxt
+ let pattern = pattern reset_ctxt
+ let signature = signature reset_ctxt
+ let structure = structure reset_ctxt
+ let module_expr = module_expr reset_ctxt
++let module_type = module_type reset_ctxt
++let class_field = class_field reset_ctxt
++let class_type_field = class_type_field reset_ctxt
++let class_expr = class_expr reset_ctxt
++let class_type = class_type reset_ctxt
++let structure_item = structure_item reset_ctxt
++let signature_item = signature_item reset_ctxt
++let binding = binding reset_ctxt
++let payload = payload reset_ctxt
+--- ocaml-4.13-upstream/pprintast.mli
++++ ocaml-4.13/pprintast.mli
+@@@@
+ val module_expr: Format.formatter -> Parsetree.module_expr -> unit
+ 
+ val toplevel_phrase : Format.formatter -> Parsetree.toplevel_phrase -> unit
+ val top_phrase: Format.formatter -> Parsetree.toplevel_phrase -> unit
+ 
++val class_field: Format.formatter -> Parsetree.class_field -> unit
```

Comment:
Have you considered upstreaming theses changes ?

Reply:
I will open a PR

Reply:
I have opened https://github.com/ocaml/ocaml/pull/10618